## 错误处理和资源管理 ##

### defer调用 ###

- 确保在函数结束时发生
- 参数在defer语句时计算
- defer列表 先进后出（栈）

### 何时使用 ###

- Open/Close
- Lock/Unlock
- PrintHeader/PrintFooter



    func tryDefer() {
		defer fmt.Print(1)
		defer fmt.Print(2)
		fmt.Print(3)
		return
		panic('error')	
		fmt.Print(4)
	}

	func tryDefer2() {
		for i:=0; i<100;i++{
			defer fmt.Println(i)
			if i ==30 {
				panic("printed too many")
			}
		}
	}

	func writeFile(filename string) {
		file,err := os.Create(filename)
		if err ！= nil {
			panic(err)
		}

		defer file.Close()
		
		writer := bufio.NewWriter(file)
		defer writer.Flush()

		f := Fibonacci()

		for i := 0; i <20; i++ {
			fmt.Fprintln(Writer,f())
		}
	}

	func main(){
		WriteFile("text.txt")
	}

	